added grid components

jenscotch opened the pull request.

jenscotch committed:
added grid components

i-am-ebere commented on src/components/login-view/login-view.jsx:
```diff
@@ -0,0 +1,65 @@
+        const data = {
+            access: name,
+            secret: password
+        };
```
why
```
const data = {
            access: name,
            secret: password
        };
        
```

Your backend is expecting :

```
const data = {
            Name: name,
            Password: password
};
```

i-am-ebere commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,79 +1,83 @@
+    })
+    .then((response) => response.json())
+    .then((data) => {
+        const movies = data.docs.map((doc) => {
```
why data.docs? your backend is returning data as an array so where does docs come from?

i-am-ebere replied:
if you comment the code from line 34 - 42 and put a `console.log(data)` you will see the structure of the value of data. i.e.
```
fetch("https://jens-movie-api.herokuapp.com/movies", {
        headers: { Authorization: `Bearer ${token}` }
    })
    .then((response) => response.json())
    .then((data) => {
       console.log(data) <-- this will show you what kind of data comes back
    })
```

i-am-ebere commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,79 +1,83 @@
+    .then((data) => {
+        const movies = data.docs.map((doc) => {
+            return {
+                id: doc.key,
```
where does key come from? do you mean `_id`?

jenscotch replied:
Buddy, you need to provide more context for me, because I don't KNOW if I meant "_id" or "id", because this is literally the way the Exercise showed me how to write this.

i-am-ebere replied:
you dont have a field called key based on this code
https://github.com/jenscotch/movie_api/blob/main/models.js
<img width="534" alt="image" src="https://github.com/jenscotch/myFlix-client/assets/13005417/9ebb6c7d-e205-40ba-9428-76a51411b534">

but mongodb creates a field called `_id` as an identifier to uniquely identify all documents in a collection (look at achievement 2.7 and https://www.mongodb.com/docs/manual/core/document/). Thats why it should be `_id`

i-am-ebere commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,79 +1,83 @@
+                id: doc.key,
+                title: doc.Title,
+                image: doc.Image,
+                director: doc.Director_Name?.[0]
```
Is this value an array or an object

jenscotch replied:
I don't know what I'm supposed to get from this comment. There are {} brackets around it, and this is the example that the Exercise provided.

i-am-ebere replied:
Its good to look at the example but your implementation isnt based on the example, it should be based on your movie_api code. 
Take a look at this file https://github.com/jenscotch/movie_api/blob/main/models.js
according this this file
<img width="534" alt="image" src="https://github.com/jenscotch/myFlix-client/assets/13005417/cdfbb896-3f51-48a2-bbcc-33da73546b3f">
 Director is an object so it should be doc.Director

jenscotch replied:
Sorry, but it IS doc.Director???? Are you saying I need to get rid of the "_Name?.[0]"?

i-am-ebere replied:
Yes thats what i am saying, based on your code from movie_api repository, line 11 - 14, Director is an object not an array. If you look at the image or the file link i posted earlier you dont have a Director_Name in your code(https://github.com/jenscotch/movie_api/blob/main/models.js).

i-am-ebere commented on src/components/main-view/main-view.jsx:
```diff
@@ -1,79 +1,83 @@
+                <Col className="mb-5" key={movie.id} md={3}>
+                <MovieCard
+                    movie={movie}
+                    onBookClick={(newSelectedMovie) => {
```
why onBookClick? the component is expecting onMovieClick?

Look at src/components/movie-card/movie-card.jsx its expecting onMovieClick
```
export const MovieCard = ({ movie, onMovieClick }) => {...}
```

i-am-ebere commented on src/components/movie-view/movie-view.jsx:
```diff
@@ -16,7 +18,13 @@ export const MovieView = ({ movie, onBackClick }) => {
                 <span>{movie.director}</span>
             </div>
-            <button onClick={onBackClick}>Back</button>
+            <Button 
```
Should you import Button from "react-bootstrap" here too?

i-am-ebere commented on src/components/movie-card/movie-card.jsx:
```diff
@@ -1,11 +1,28 @@
-            {movie.title}
-        </div>
+        <Card className="h-100">
+            <Card.Img variant="top" src={movie.image} />
```
You api doesnt return image. This is a sample response from your api

```
[
{ 
Description : 
"A struggling mother of three forms an unexpected bond with the night nanny hired to help with her newborn baby."
Director:  { Name: 'Jason Reitman', Bio: 'Jason Reitman is a Canadian  filmmaker and produce…ected the first two Ghostbusters films and Twins.', Birth: '1977'},
Featured : true,
Genre: { Name: 'Drama', Description: 'In film and television, drama is a category or gen…nd its resolution in the course of the storyline.'},
Title: "Tully",
_id: "644ebf0575f1a072ec75018b"
}
...
]
```

jenscotch committed:
fixed bugs

fixed display bugs in main-view, movie_card, & movie-view

jenscotch committed:
addition of navbar, profile view, user info

added components for a navigation bar, profile view, and user info/capabilities

jenscotch committed:
trying to set fav movie

more attempts at trying to set user's favorite movie in movie-card

jenscotch committed:
.map changes

jenscotch committed:
Merge branch 'main' into 4
